### DISCUSSION TITLE
Erro '403-Forbidden for url ...'

### DISCUSSION THREAD

**Discussion by fbminfo** [qualitaocupacional/libesocial#5] [Q&A] [answered]
Boa tarde.
Estou fazendo testes com essa lib (maravilhosa - diga-se de passagem), mas está retornando  o erro "403 Client Error: Forbidden for url: https://webservices.producaorestrita.esocial.gov.br/servicos/empregador/enviarloteeventos/WsEnviarLoteEventos.svc?singleWsdl".
Até semana passada esse erro raramente aparecia. Agora tá aparecendo sempre.
Alguém poderia me auxiliar, por gentileza?
Obrigado,
Francisco.

**Comment by andidevel** [comment, marked as answer]
Olá Francisco, que legal que você está usando a lib e está gostando.

Quanto a sua questão, não sei se você está alterando alguma URL padrão, mas como já está configurado no `__init__.py` do módulo `esocial`, a URL correta para conexão é:

`https://webservices.producaorestrita.esocial.gov.br/servicos/empregador/enviarloteeventos/WsEnviarLoteEventos.svc?wsdl`.

Entretanto, geralment os erros "**403**", são causados por certificados digitais do *transmissor* ou *empregador* inválidos, vencidos ou auto-assinados.

Também é possível que os servidores de **Produção Restrita** estejam passando por alguma manutenção. Até o momento não encontramos nenhum problema no servidor oficial (*Produção Empresas*).

**Comment by fbminfo** [comment]
Boa tarde Anderson.

Te agradeço imensamente pela atenção.

Eu também testei a URL que você mencionou e, depois de centenas de testes,
acabei optando pela

https://webservices.producaorestrita.esocial.gov.br/servicos/empregador/enviarloteeventos/WsEnviarLoteEventos.svc?singleWsdl
,

que peguei nesse site:
https://webservices.producaorestrita.esocial.gov.br/servicos/empregador/enviarloteeventos/WsEnviarLoteEventos.svc,
mas não me lembro mais do motivo.

Os erros "403" aparecem na duas URL's.

Os certificados usados são válidos. Acredito que seja problema nos
servidores de Produção Restrita mesmo, como você mencionou, pois venho
fazendo muitos testes diários com 'TpAmb=1' e até agora não apareceu o erro
'403'.

Mais uma vez, muito obrigado pela atenção e parabéns pelo seu excelente
trabalho.

[]'s
Francisco.



Em seg., 8 de ago. de 2022 às 13:34, Anderson R. Livramento <
***@***.***> escreveu:

> Olá Francisco, que legal que você está usando a lib e está gostando.
>
> Quanto a sua questão, não sei se você está alterando alguma URL padrão,
> mas como já está configurado no __init__.py do módulo esocial, a URL
> correta para conexão é:
>
>
> https://webservices.producaorestrita.esocial.gov.br/servicos/empregador/enviarloteeventos/WsEnviarLoteEventos.svc?wsdl
> .
>
> Entretanto, geralment os erros "*403*", são causados por certificados
> digitais do *transmissor* ou *empregador* inválidos, vencidos ou
> auto-assinados.
>
> Também é possível que os servidores de *Produção Restrita* estejam
> passando por alguma manutenção. Até o momento não encontramos nenhum
> problema no servidor oficial (*Produção Empresas*).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qualitaocupacional/libesocial/discussions/5#discussioncomment-3351100>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZJ6CJNHIZCAAOMVNGV5YCDVYEZIJANCNFSM55W7UCBA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
==================
BM Informatica
***@***.***
***@***.***
<* Powered By Linux *>
